### PR TITLE
disable linking mvsc static runtime in treelite

### PR DESCRIPTION
**Pull request by harishneit**
Thanks for contributing to DLR! By submitting this pull request, you confirm that your contribution is made under the terms of the [Apache 2.0 license](https://github.com/neo-ai/neo-ai-dlr/blob/master/LICENSE).

Please refer to our [guideline](https://github.com/neo-ai/neo-ai-dlr/blob/master/CONTRIBUTING.md) for useful information and tips.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -372,7 +372,6 @@ set(NEODLR_SRC ${DLR_SRC})
 list(REMOVE_ITEM NEODLR_SRC 
     ${TVM_SRC}/src/runtime/dso_library.cc
     ${TVM_SRC}/src/runtime/cpu_device_api.cc
-    ${TVM_SRC}/src/runtime/metadata_module.cc
```
Comment (mycpuorg) [inline CMakeLists.txt:375]: ?

File: cmake/neodlrInstall.cmake [hunk 2]
```diff
@@ -16,7 +16,7 @@ endforeach()
 
 set(NEODLR_HEADERS include/dlr.h)
 
-set_target_properties(neodlr PROPERTIES PUBLIC_HEADER "include/dlr.h;include/dlr_tvm.h;include/dlr_treelite.h")
+set_target_properties(neodlr PROPERTIES PUBLIC_HEADER ${NEODLR_HEADERS})
```
Comment (mycpuorg) [inline cmake/neodlrInstall.cmake:19]: Won't this break any client applications dep on these removed headers?

Comment (harishneit) [reply cmake/neodlrInstall.cmake:19]: we only depend on dlr.h

File: CMakeLists.txt (hunk 1 above)
Comment (harishneit) [reply CMakeLists.txt:375]: essentially we trying to remove lines 107-110. this got added by mistake while cherrypicking but it is redundant. hence remove it in the patch.

**Comment by trevor-m** [comment]
Hey @harishneit we have to make sure dmlc-core is synced between treelite, TVM, and DLR. Please update all of them. We will need to test a bit more to see if the TVM 1.2 compiled models are compatible with the dmlc-core you want to use.

**Comment by harishneit** [comment]
hey @trevor-m this is reverted in a new patch by jason

**Pull request closed without merging**